src/rust/graph-query-service/tests/integration_test.rs

```diff
@@ -1 +1,255 @@
+                    source_uid bigint,
+                    destination_uid bigint,
+                    f_edge_name text,
+                    r_edge_name text,
+                    PRIMARY KEY ((source_uid, f_edge_name), destination_uid)
+                )"
+            ),
+            &(),
+        )
+        .await?;
+
+    session.session.await_schema_agreement().await?;
+
+    Ok(tenant_ks)
+}
+
+async fn get_scylla_client() -> Result<Arc<CachingSession>, DynError> {
+    let graph_db_config = GraphDbConfig::parse();
+
+    let mut scylla_config = scylla::SessionConfig::new();
+    scylla_config.add_known_nodes_addr(&graph_db_config.graph_db_addresses[..]);
+    scylla_config.auth_username = Some(graph_db_config.graph_db_auth_username.to_owned());
+    scylla_config.auth_password = Some(
+        graph_db_config
+            .graph_db_auth_password
+            .expose_secret()
+            .to_owned(),
+    );
+
+    let scylla_client = Arc::new(CachingSession::from(
+        scylla::Session::connect(scylla_config).await?,
+        10_000,
+    ));
+    Ok(scylla_client)
+}
+
+#[tracing::instrument(skip(session), fields(
+    keyspace = %keyspace.as_ref(),
+    uid=%uid.as_i64(),
+    node_type=node_type.value.as_str(),
+) err)]
+async fn create_node(
+    session: &CachingSession,
+    keyspace: impl AsRef<str>,
+    uid: Uid,
+    node_type: &NodeType,
+) -> Result<(), DynError> {
+    let keyspace = keyspace.as_ref();
+    let insert = format!(
+        r"INSERT INTO {keyspace}.node_type
+                           (uid, node_type)
+                           VALUES (?, ?)"
+    );
+
+    session
+        .execute(insert, &(uid.as_i64(), &node_type.value))
+        .await?;
+
+    Ok(())
+}
+
+async fn insert_string(
+    session: &CachingSession,
+    keyspace: impl AsRef<str>,
+    uid: Uid,
+    populated_field: impl AsRef<str>,
+    value: impl AsRef<str>,
+) -> Result<(), DynError> {
+    let keyspace = keyspace.as_ref();
+    let insert = format!(
+        r"INSERT INTO {keyspace}.immutable_strings
+                           (uid, populated_field, value)
+                           VALUES (?, ?, ?)"
+    );
+
+    session
+        .execute(
+            insert,
+            &(uid.as_i64(), populated_field.as_ref(), value.as_ref()),
+        )
+        .await?;
+
+    Ok(())
+}
+
+#[test_log::test(tokio::test)]
+async fn test_query_single_node() -> Result<(), DynError> {
+    let _span = tracing::info_span!(
+        "tenant_id", tenant_id=?tracing::field::Empty,
+    );
+    tracing::info!("starting test_query_single_node");
+    let graph_query_service_endpoint = std::env::var("GRAPH_QUERY_SERVICE_ENDPOINT_ADDRESS")
+        .expect("GRAPH_QUERY_SERVICE_ENDPOINT_ADDRESS");
+    tracing::info!(
+        graph_query_service_endpoint=%graph_query_service_endpoint,
+        message="connecting to graph query service"
+    );
+    let mut graph_query_client = GraphQueryClient::connect(graph_query_service_endpoint).await?;
```

Comment:
This is different from the rest of our codebase, where we instead prefer something like
```
        let client_config = PluginRegistryClientConfig::parse();
        let plugin_registry_client = build_grpc_client(client_config).await?;
```

Could you add a GraphQueryClientConfig to src/rust/rust-proto/src/client_factory/services ?